From d417956eb852b92befdde5e56e546616f9247195 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 07 Aug 2018 11:19:56 +0300
Subject: [PATCH] board: habana: remote_run: consolidate the arguments

Consolidate all the arguments that should be passed to the local_run script
into a single variable delimited with spaces.

Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/board/habanalabs/common/remote_run b/board/habanalabs/common/remote_run
index 1fdd217..d333ac3 100755
--- a/board/habanalabs/common/remote_run
+++ b/board/habanalabs/common/remote_run
@@ -23,6 +23,8 @@
 	exit 10;
 fi
 
+remote_args="$remove_pkgs $install_pkgs $program_flash"
+
 function remote_prepare_work_dirs() {
 	ssh -t $host "sudo mkdir -p $fw_dir;" | tee -a $result
 
@@ -48,9 +50,9 @@
 }
 
 function remote_run_script() {
-	print_msg "Starting ${scr_name[0]}, args: $remove_pkgs $install_pkgs $program_flash" | tee -a $result
+	print_msg "Starting ${scr_name[0]}, args: $remote_args" | tee -a $result
 
-	ssh -t $host "${remote_scr_dir}/${scr_name[0]} $remove_pkgs $install_pkgs $program_flash;" | tee -a $result
+	ssh -t $host "${remote_scr_dir}/${scr_name[0]} $remote_args;" | tee -a $result
 
 	local pipestat=${PIPESTATUS[0]}
 	case $pipestat in
@@ -68,7 +70,7 @@
 		;;
 	esac | tee -a $result
 
-	print_msg "${scr_name[0]} script status: $pipestat" | tee -a $result
+	print_msg "${scr_name[0]} ($remote_args) script status: $pipestat" | tee -a $result
 	return $pipestat;
 }
 
